packages/analytics-js/src/services/ErrorHandler/ErrorHandler.ts

```diff
@@ -29,6 +30,19 @@ class ErrorHandler implements IErrorHandler {
     this.pluginEngine = pluginEngine;
   }
 
+  attachErrorListeners() {
+    if ('addEventListener' in (globalThis as any)) {
+      (globalThis as any).addEventListener('error', (event: any) => {
+        this.onError(event, undefined, undefined, undefined, 'unhandledException');
+      });
+    }
+    if ('addEventListener' in (globalThis as any)) {
+      (globalThis as any).addEventListener('unhandledrejection', (event: any) => {
+        this.onError(event.reason, undefined, undefined, undefined, 'unhandledPromiseRejection');
```

Comment:
I am not sure passing only the event.reason here is correct

Reply:
You probably want to pass the event here,a nd then in processError make use of reason instead of message

Reply:
Addressed